File: src/p1002/python/solution.py
from typing import List
import math


class Solution:
    def commonChars(self, ss: List[str]) -> List[str]:
        curr = [math.inf] * 26
        for s in ss:
            next = [0] * 26
            for c in s:
                if curr[ord(c) - ord('a')] > 0:
                    curr[ord(c) - ord('a')] -= 1
                    next[ord(c) - ord('a')] += 1
            curr = next
        result = []
        for i, v in enumerate(curr):
            for _ in range(v):
                result.append(chr(i + ord('a')))
        return result

File: src/p1002/golang/solution.go
package main

import (
	"fmt"
)

func commonChars(strs []string) (result []string) {
	var curr, next [26]int
	for i := 0; i < len(strs); i++ {
		for j := 0; j < len(strs[i]); j++ {
			c := strs[i][j]
			if i == 0 || curr[c - 'a'] > 0 {
				curr[c - 'a']--
				next[c - 'a']++
			}
		}
		for k := 0; k < 26; k++ {
			curr[k], next[k] = next[k], 0
		}
	}
	for i, v := range curr {
		for ; v > 0; v-- {
			result = append(result, fmt.Sprintf("%c", i + 'a'))
		}
	}
	return result
}
